scripts/clippy.sh

```diff
@@ -32,6 +44,11 @@ if [[ "$OSTYPE" == "linux-gnu"* ]]; then
    )
 fi
 
+# Do not use --all-features for the following projects
+NO_ALL_FEATURES=(
```

Comment:
I'm still not a big fan of this solution, IMHO the clippy feature / --all-features should just make CI work?

Reply:
(I guess we discussed this before but I forgot / still don't understand the reasoning against it)

Reply:
The reason was:
> The thing is that in `libafl_qemu` some features are incompatible (`usermode` and `systemmode` for instance), running with `--all-features` should fail.
> 
> So far the build script decided which feature to actually use and which silently ignore when running with `--all-features`. In my opinion having `--all-features` just working but actually not enabling all features is not ideal. Looking at the clippy script it looks like every feature of libafl_qemu is checked with clippy while it is not.

IMHO the default features should _just work™_, not the `--all-features` here.

Also, in practice, this change allows the usage of optional dependencies

Reply:
100% agree about the clippy feature tho